middleware/compress.go

```diff
@@ -102,18 +138,52 @@ func GzipWithConfig(config GzipConfig) echo.MiddlewareFunc {
 
 func (w *gzipResponseWriter) WriteHeader(code int) {
 	w.Header().Del(echo.HeaderContentLength) // Issue #444
-	w.ResponseWriter.WriteHeader(code)
+
+	w.wroteHeader = true
+
+	// Delay writing of the header until we know if we'll actually compress the response
```

Comment:
I think this breaks `http.ResponseWriter.WriteHeader` interface behavior. https://github.com/golang/go/blob/0853f8caec60f59df234c287be7f5971ab62133f/src/net/http/server.go#L159 Nothing says that you could not have handler that only calls `WriteHeader` and does not call `Write`  - in that case gzipResponseWriter ends without sending status.

Reply:
my bad. there is that 
```go
					} else if !grw.minLengthExceeded {
						// Write uncompressed response
						res.Writer = rw
						if grw.wroteHeader {
							grw.ResponseWriter.WriteHeader(grw.code)
						}
```
block that writes status